src/main/java/com/minecolonies/client/gui/WindowHutFarmer.java

```diff
@@ -1,243 +1,316 @@
-     *                  This method does not
+     * @param building {@link com.minecolonies.colony.buildings.BuildingFarmer.View}
      */
-    private void potatoClicked(Button ignored)
+    public WindowHutFarmer(BuildingFarmer.View building)
     {
-        if (building.potato >= MAX_AMOUNT)
-        {
-            return;
-        }
-        building.potato++;
-        removeOthers(POTATO);
+        super(building, Constants.MOD_ID + HUT_FARMER_RESOURCE_SUFFIX);
+        registerButton(BUTTON_PREV_PAGE, this::prevClicked);
+        registerButton(BUTTON_NEXT_PAGE, this::nextClicked);
+        registerButton(TAG_BUTTON_ASSIGNMENT_MODE, this::assignmentModeClicked);
+        registerButton(TAG_BUTTON_ASSIGN, this::assignClicked);
     }
 
     /**
-     * Action performed when melon button is clicked
+     * Fired when assign has been clicked in the field list.
      *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * @param button clicked button.
      */
-    private void melonClicked(Button ignored)
+    private void assignClicked(@NotNull Button button)
     {
-        if (building.melon >= MAX_AMOUNT)
+        final int row = fieldList.getListElementIndexByPane(button);
+        final FieldView field = fields.get(row);
+
+
+        if (button.getLabel().equals(RED_X))
         {
-            return;
+            button.setLabel(APPROVE);
+            building.changeFields(field.getId(), false, row);
+        }
+        else
+        {
+            button.setLabel(RED_X);
+            building.changeFields(field.getId(), true, row);
         }
-        building.melon++;
-        removeOthers(MELON);
-    }
 
+        pullLevelsFromHut();
+        window.findPaneOfTypeByID(LIST_FIELDS, ScrollingList.class).refreshElementPanes();
+    }
 
     /**
-     * Action performed when pumpkin button is clicked
-     *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * Retrieve levels from the building to display in GUI
      */
-    private void pumpkinClicked(Button ignored)
+    private void pullLevelsFromHut()
     {
-        if (building.pumpkin >= MAX_AMOUNT)
-        {
-            return;
-        }
-        building.pumpkin++;
-        removeOthers(PUMPKIN);
+        fields = building.getFields();
     }
 
-
     /**
-     * Action performed when carrot button is clicked
+     * Fired when the assignment mode has been toggled.
      *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * @param button clicked button.
      */
-    private void carrotClicked(Button ignored)
+    private void assignmentModeClicked(@NotNull Button button)
     {
-        if (building.carrot >= MAX_AMOUNT)
+        if (button.getLabel().equals(LanguageHandler.format("com.minecolonies.gui.hiring.off")))
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "com.minecolonies.gui.hiring.off" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)